File: packages/ios/CoreKit/Sources/CoreKit/APIClient/APIEndpoint.swift
import Foundation

/// API 端点定义
public enum APIEndpoint: Sendable {
    case login
    case register
    case logout
    case refreshToken
    case chat
    case chatHistory
    case appendChatHistory
    case analyze
    case grade
    case plan
    case planHistory
    case updatePlanTask(String)
    case profile
    case updateProfile
    case gradeHistory
    case learnerProfile
    case upload
    case uploadPresign
    case wrongQuestions
    case reviewWrongQuestion
    case addWrongQuestion
    case updateAnalysisBookmark(String)
    case stats
    case bankCount
    case bank
    case submitBankPractice

    public var path: String {
        switch self {
        case .login:               return "/api/auth/login"
        case .register:            return "/api/auth/register"
        case .logout:              return "/api/auth/logout"
        case .refreshToken:        return "/api/auth/refresh"
        case .chat:                return "/api/chat"
        case .chatHistory:         return "/api/chat/history"
        case .appendChatHistory:   return "/api/chat/history"
        case .analyze:             return "/api/analyze"
        case .grade:               return "/api/grade"
        case .plan:                return "/api/plan"
        case .planHistory:         return "/api/plan"
        case .updatePlanTask(let taskId): return "/api/plan/tasks/\(taskId)"
        case .profile, .updateProfile: return "/api/profile"
        case .gradeHistory:        return "/api/grade/history"
        case .learnerProfile:      return "/api/learner-profile"
        case .upload:              return "/api/upload"
        case .uploadPresign:       return "/api/upload/presign"
        case .wrongQuestions:      return "/api/wrong-questions"
        case .reviewWrongQuestion: return "/api/wrong-questions"
        case .addWrongQuestion:    return "/api/wrong-questions/add"
        case .updateAnalysisBookmark(let questionId): return "/api/analysis/\(questionId)"
        case .stats:               return "/api/stats"
        case .bankCount:           return "/api/bank/count"
        case .bank:                return "/api/bank"
        case .submitBankPractice:  return "/api/bank/practice"
        }
    }

    public var method: String {
        switch self {
        case .wrongQuestions, .stats, .bankCount, .bank, .chatHistory, .planHistory, .profile, .gradeHistory, .learnerProfile:
            return "GET"
        case .updateProfile, .updatePlanTask(_), .updateAnalysisBookmark(_):
            return "PATCH"
        default:
            return "POST"
        }
    }
}

File: packages/android/corekit/src/main/kotlin/com/aistudy/corekit/net/ApiEndpoint.kt
package com.aistudy.corekit.net

/**
 * API 端点路径常量（镜像 iOS `APIEndpoint.swift`）。
 * 集中管理便于核对与 iOS / Web route 一致。Retrofit 注解里直接写字符串，这里作参考与单测断言用。
 */
object ApiEndpoint {
    const val LOGIN = "api/auth/login"
    const val REGISTER = "api/auth/register"
    const val LOGOUT = "api/auth/logout"
    const val REFRESH = "api/auth/refresh"
    const val CHAT = "api/chat"
    const val CHAT_HISTORY = "api/chat/history"
    const val ANALYZE = "api/analyze"
    const val GRADE = "api/grade"
    const val PLAN = "api/plan"
    const val UPLOAD = "api/upload"
    const val UPLOAD_PRESIGN = "api/upload/presign"
    const val WRONG_QUESTIONS = "api/wrong-questions"
    const val REVIEW_WRONG_QUESTION = "api/wrong-questions"
    const val STATS = "api/stats"
    const val BANK_COUNT = "api/bank/count"
}
